dbldatagen/utils.py

```diff
@@ -214,3 +215,57 @@ def parse_time_interval(spec):
     )
 
     return delta
+
+
+def split_list_matching_condition(lst, cond):
+    """ Split a list on elements that match a condition
+
+    This will find all matches of a specific condition in the list and split the list into sublists around the
+    element that matches this condition.
+
+    It will handle multiple matches performing splits on each match.
+
+    For example, the following code will produce the results below:
+
+    x = ['id', 'city_name', 'id', 'city_id', 'city_pop', 'id', 'city_id', 'city_pop','city_id', 'city_pop','id']
+    splitListOnCondition(x, lambda el: el == 'id')
+
+
+    result:
+    `[['id'], ['city_name'], ['id'], ['city_id', 'city_pop'],
+     ['id'], ['city_id', 'city_pop', 'city_id', 'city_pop'], ['id']]`
+
+    :arg lst: list of items to perform condition matches against
+    :arg cond: lambda function or function taking single argument and returning True or Fals
+    :returns: list of sublists
+    """
+
+    def match_condition(matchList, matchFn):
+        """Return first index of element of list matching condition"""
+        if matchList is None or len(matchList) == 0:
```

Comment:
this is redundant due to the if statement on line 257

Reply:
Not quite correct - consider case where len(matchlist) == 0

Outer if does not check for it. 

In any event want inner `match_condition` to guard against empty list 

We could drop outer check but cant take len(None)